model-optimizer/unit_tests/extensions/back/add_outputs_recursive_test.py

```diff
@@ -566,6 +566,137 @@ def test_add_output_several_iterations_negative_stride(self):
         self.assertEqual(unsq_node.out_port(0).get_destination().node.internal_layer_id, max_layer_id + 3)
         self.assertTrue(np.all(unsq_node.out_port(0).data.get_shape() == int64_array([1, 1, 4, 64, 54])))
 
+    def test_add_output_several_iterations_negative_start_end_input(self):
+        sub_graph_2 = build_graph(nodes_attrs=ti_sub_graph_2_nodes,
+                                  edges=[*connect('cond_2_int', 'cond_2_int_out'),
+                                         *connect('in_2_int', 'OUT_2'),
+                                         *connect('ones', 'OUT_2'),
+                                         *connect('OUT_2', 'OUT_2_out'),
+                                         *connect('in_2_int', 'in_2_int_out')],
+                                  nodes_with_edges_only=True)
+
+        sub_graph_1 = build_graph(nodes_attrs=ti_sub_graph_1_nodes,
+                                  edges=[*connect('cond_2', '1:Loop_2'),
+                                         *connect('IN_2', '0:Loop_2'),
+                                         *connect('Loop_2:0', 'Loop_2_out'),
+                                         *connect('in_1_int', 'in_1_int_out'),
+                                         *connect('cond_1_int', 'cond_1_int_out')],
+                                  nodes_with_edges_only=True)
+        loop_node_1 = Node(sub_graph_1, 'Loop_2')
```

Comment:
please create common routine for graph creation. Now we have a lot of duplicates here